submissions/Complexity Corner - The Mall/Complexity Corner - The Mall.md

```diff
@@ -0,0 +1,17 @@
+### Complexity Corner: The Mall <author>The Kool</author>
+
+Welcome back once more to the Complexity Corner.  Today I'll be exploring a technique instead of a technology: the Mall.  As always with these articles, we'll be asking three questions: 'What is it?', 'What results can it give?', and 'Where should it be used?'
+
+What is this 'Mall' I keep hearing about?
+
+We've all experienced it by now, setting up something to craft manually and then drumming your fingers on the desk while you wait for the mile-long queue to finish.  Wouldn't it be so much nicer to walk over to a chest and have some waiting for you?  To just go shopping for your next project, instead of crafting by hand?  That's what the Mall is for.  Without even realizing it, you probably only think of automating things that have an automated end goal, such as ammo for your turrets or science for your labs.  The Mall is a place where you automate things that have a manual end goal - You.
+
+So what's the benefit?
+
+A typical early-game smelting line consists of 24 stone furnaces if you're planning ahead to upgrade them.  In order to feed these furnaces, you need 48 inserters, 24 long-handed inserters, and ~100 belts.  If you carry around only raw materials like plates and craft these things as needed, that's 12s for the furnaces, 50s for the belts, and 93s for the inserters; a total of 2m 35s that you're waiting around before you can build that one furnace line.  Multiply that by every furnace line, every mining outpost, and every factory cell you build and you'll soon find you're standing around waiting far more than you thought.  The mall makes that go away, building things for you while you build your factory.
+
+Alright, when and where do I use it?
+
+The Mall can (and should!) be used as early as you can in the game.  Your first instinct may be to automate red science to the lab, but I encourage you to first build a simple mall as soon as you have assembly machines.  Have it build just gears and circuits at first, but soon upgrade it to making belts, inserters, and assembly machines.  Those three items and two ingredients will cover the majority of what you'll be making for the first stage of the game, and all you need to do is come back and add on when there's something new you'll be using a bunch of, like fast inserters or advanced circuits.
```

Comment:
Another interesting tidbit you could add is to have two output chests for each product in the mall. One to pickup from with control click containing a small amount, then a second chest containing a large supply should you need a large quantity.